src/util/initEvents.ts

```diff
@@ -16,16 +16,25 @@ declare global {
 /** Add window event handlers. */
 export const initEvents = (store: Store<State, any>) => {
 
+  // eslint-disable-next-line fp/no-let
+  let lastState: number
+
   /** Popstate event listener; setCursor on browser history forward/backward. */
-  const onPopstate = () => {
+  const onPopstate = (e: PopStateEvent) => {
 
     const state = store.getState()
 
-    const { path, contextViews } = decodeThoughtsUrl(state, window.location.pathname)
+    const { path, contextViews } = decodeThoughtsUrl(state, window.location.pathname, { exists: true })
```

Comment:
If you add `exists: true`, then `path` will be `null` in the case of a nonexistent pathname and you won't be able to differentiate it from the root context, as described here:

> It seems that `decodeThoughtsUrl` is a little inconsistent; when `exists: true` is given it returns `null` for a valid root context. This might be a bug.

https://github.com/cybersemics/em/issues/884#issuecomment-734508411

Not sure if that was intended, but I think with the current implementation of `decodeThoughtsUrl` you'll have to omit `exists` and use `pathExists` directly.

Reply:
@raineorshine Removing `exists:true` solves the problem. You can test the functionality now.
As discussed, I'll look into the `decodeThoughtsUrl` bug and it's relation to the state in a different issue.